### PR TITLE
Added connection state observable

### PR DESCRIPTION
**Pull request by robholmes**
I've added an observe method to allow monitoring of device connection states. Essentially just a broadcast receiver for `BluetoothAdapter.ACTION_CONNECTION_STATE_CHANGED`.

### CODE REVIEW

File: rxbluetooth/src/main/java/com/github/ivbaranov/rxbluetooth/ConnectionStateEvent.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    return mBluetoothDevice;
+  }
+
+  @Override public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+
+    ConnectionStateEvent that = (ConnectionStateEvent) o;
+
+    if (mState != that.mState) return false;
```
Comment (IvBaranov) [inline rxbluetooth/src/main/java/com/github/ivbaranov/rxbluetooth/ConnectionStateEvent.java]: mPreviousState check is missing

Comment (robholmes) [inline rxbluetooth/src/main/java/com/github/ivbaranov/rxbluetooth/ConnectionStateEvent.java]: thanks for spotting that, will fix right away :+1:

Comment (robholmes) [inline rxbluetooth/src/main/java/com/github/ivbaranov/rxbluetooth/ConnectionStateEvent.java]: @IvBaranov fixed :smiley:

**Comment by IvBaranov** [comment]
LGTM!
Thanks for you contribution!

**Pull request merged**